### ISSUE TITLE
Use in Azure Functions

### ISSUE THREAD

**Issue by igor-pikovsky** [micro-elements/MicroElements.Swashbuckle.FluentValidation#62]
Is there a way to use this with REST API implemented using Azure Functions (serverless)?
Thank you

**Comment by petriashev** [comment]
MicroElements.Swashbuckle.FluentValidation relies on Swashbuckle.AspNetCore infrastructure

I found project Swashbuckle.AspNetCore.AzureFunctions that generates OpenApi declaration with Swashbuckle.AspNetCore.

Plan:
1. First you should plug Swashbuckle.AspNetCore.AzureFunctions into your project
2. Then try add 'c.AddFluentValidationRules();' in services.AddSwaggerGen and it should work. (probably)

**Comment by igor-pikovsky** [comment]
Unfortunately the project Swashbuckle.AspNetCore.AzureFunctions has been discontinued and removed from GitHub

**Comment by igor-pikovsky** [comment]
In addition to the above.
I can include AddFluentValidationRules - that is not a problem.

The problem is with the call
```
services
        .AddControllers()
        // Adds fluent validators to Asp.net
        .AddFluentValidation(c =>
        {
            c.RegisterValidatorsFromAssemblyContaining<Startup>();
            // Optionally set validator factory if you have problems with scope resolve inside validators.
            c.ValidatorFactoryType = typeof(HttpContextServiceProviderValidatorFactory);
        })
```
since in Azure Functions there is no call AddControllers - as there are no explicit controllers.

**Comment by petriashev** [comment]
I will try to make project less host dependent to support NSwag, AzureFunctions

**Comment by tommck** [comment]
So this currently doesn't work with Azure Functions? :(

**Comment by petriashev** [comment]
@tommck Yes, Azure Functions is not AspNetCore app. Swagger works only with AspNetCore now.

**Comment by tommck** [comment]
Swagger works with Azure Functions using this: https://github.com/vitalybibikov/AzureExtensions.Swashbuckle

That's why I asked

**Comment by tommck** [comment]
Has there been any progress on this @petriashev?

**Comment by petriashev** [comment]
@tommck I had not possibility to develop such big things.

**Comment by petriashev** [comment]
I've done some investigation about this issue. I should untie infrastructure code from AspNet, Swashbuckle. It will be possible after issue #63
